src/hotspot/share/memory/virtualspace.hpp

```diff
@@ -58,26 +58,26 @@ class ReservedSpace {
   //                       0 during initialization.
   void clear_members();
   void initialize_members(char* base, size_t size, size_t alignment,
-                          size_t page_size, bool special, bool executable);
+                          size_t page_size, bool special, bool executable, MemTag mem_tag);
 
   void initialize(size_t size, size_t alignment, size_t page_size,
-                  char* requested_address, bool executable);
+                  char* requested_address, bool executable, MemTag mem_tag);
 
   void reserve(size_t size, size_t alignment, size_t page_size,
-               char* requested_address, bool executable);
+               char* requested_address, bool executable, MemTag mem_tag);
```

Comment:
Why do the instance methods get MemTag? Why is MemTag not stored as a (const) attribute of ReservedSpace? 

But then, we would have double accounting - we'd store the tag both in NMT and in ReservedSpace instances. If we change them post-reservation, we would need to change both. Sigh.

So we pass in MemTag just for the ReservedSpace to pass it into os::reserve_memory. Seeing how often ReservedSpace is used in situations where the tag is not clear at reservation time, I wonder whether we are not better off letting the user of ReservedSpace register the NMT tag post reservation like they do today.

Reply:
I like the simplicity of not having to deal with changing the tag afterwards personally. Does anyone else have an opinion here?

Reply:
We need to keep a member MemTag for ReservedSpace. There are cases in the code where a new copy of a ReservedSpace is to be created. Then, we need to know the original MemTag and pass it again down to the other functions.